web/source/osk/visualKeyboard.ts

```diff
@@ -1675,26 +1576,125 @@ namespace com.keyman.osk {
       }
     }
 
-  /**
-   * Touch hold key display management
-   *
-   * @param   {Object}  key   base key object
-   */
-  touchHold(key: KeyElement) {
-    // Clear and restart the popup timer
-    if(this.subkeyDelayTimer) {
-      window.clearTimeout(this.subkeyDelayTimer);
-      this.subkeyDelayTimer = null;
+    /**
+     * Starts an implementation-specific longpress gesture.  Separately implemented for
+     * in-browser and embedded modes.
+     * @param key The base key of the longpress.
+     * @returns 
+     */
+    startLongpress(key: KeyElement): PendingGesture {
+      let _this = this;
+
+      // First-level object/Promise:  will produce a subkey popup when the longpress gesture completes.
+      // 'Returns' a second-level object/Promise:  resolves when a subkey is selected or is cancelled.
+      let pendingLongpress = new browser.PendingLongpress(this, key);
+      pendingLongpress.promise.then(function(subkeyPopup) {
+        // In-browser-specific handling.
+        if(subkeyPopup) {
+          // Append the touch-hold (subkey) array to the OSK
+          let keyman = com.keyman.singleton;
+          keyman.osk._Box.appendChild(subkeyPopup.element);
+          keyman.osk._Box.appendChild(subkeyPopup.shim);
+
+          // Must be placed after its `.element` has been inserted into the DOM.
+          subkeyPopup.reposition(_this);
+        }
+      });
```

Comment:
Source: https://github.com/keymanapp/keyman/blob/759f3be852be7bc78c35b6a37748691cf3ee5b3f/web/source/osk/visualKeyboard.ts#L1093-L1112

The key tip part is handled within the `Promise` defined within `initGestures` below, which handles both 'in-browser' and 'embedded' longpresses.